### ISSUE TITLE
Add `initialFocus` to Drawer similar to Dialog

### ISSUE THREAD

**Issue by origami-z** [jpmorganchase/salt-ds#5058]
Dialog has `initialFocus` to set focus upon open. Drawer should be able to do the same.

Site example & accessibility doc should be updated as well. e.g., if no close button is used, what's best practices around initial focus? First of form element, or drawer `div`?

[SO 102130](https://go/stack/questions/102130)
